server/src/test/java/org/opensearch/index/query/ExistsQueryBuilderTests.java

```diff
@@ -89,9 +89,14 @@ protected ExistsQueryBuilder doCreateTestQueryBuilder() {
         }
         // also sometimes test wildcard patterns
         if (randomBoolean()) {
-            if (randomBoolean()) {
```

Comment:
I think the original intent of having 2 `randomBoolean()` was to reduce the probability of using a wild-card pattern to `0.25` which is implied by the comment `// also sometimes test wildcard patterns`. 

Logically equivalent logic will be:

```
if (randomBoolean() && randomBoolean()) {
....wild-card related logic....
} 
```

Now, honestly I don't know the reason behind having wild-card patters `sometimes` instead of just randomly 50% of the time, and don't have arguments to decide either way. However, let's change the behavior only if we intend to, else let's keep it the way it is.

Reply:
Good catch. You're right about the 25% probability. I've restored the nested randomBoolean() structure to preserve the original behavior.

 I agree we should only change the probability if there's a specific reason to.